src/BotwinExtensions.cs

```diff
@@ -35,23 +37,58 @@ public static IApplicationBuilder UseBotwin(this IApplicationBuilder builder, Bo
             ApplyGlobalAfterHook(builder, options);
 
             var routeBuilder = new RouteBuilder(builder);
+            var systemRoutes = new List<(string verb, string route)>();
 
             //Create a "startup scope" to resolve modules from
             using (var scope = builder.ApplicationServices.CreateScope())
             {
+                var modules = scope.ServiceProvider.GetServices<BotwinModule>();
+
                 //Get all instances of BotwinModule to fetch and register declared routes
-                foreach (var module in scope.ServiceProvider.GetServices<BotwinModule>())
+                foreach (var module in modules)
                 {
                     var moduleType = module.GetType();
 
                     foreach (var route in module.Routes.Keys)
                     {
                         routeBuilder.MapVerb(route.verb, route.path, CreateRouteHandler(route, moduleType));
+
+                        var strippedPath = route.path.EndsWith("/") ? route.path.Substring(0, route.path.Length - 1) : route.path;
+                        systemRoutes.Add((route.verb, "/" + strippedPath));
                     }
                 }
+
+                builder.UseRouter(routeBuilder.Build());
+
+                return builder.Use((ctx, next) => GetMethodNotAllowedHandler(ctx, next, systemRoutes));
             }
+        }
+
+        /// <summary>
+        /// This method is only called if it's a valid 404 or 405
+        /// </summary>
+        /// <param name="context"></param>
+        /// <param name="next"></param>
+        /// <param name="systemRoutes"></param>
+        /// <returns></returns>
+        private static async Task GetMethodNotAllowedHandler(HttpContext context, Func<Task> next, IEnumerable<(string verb, string route)> systemRoutes)
```

Comment:
Since this does not return a handler function (which is what I was originally suggesting) but *is* the handler, the `Get` feels out of place here. `HandleMethodNotAllowed`? Or maybe you should just move this out to a custom middleware class.

Reply:
Yeah tried your approach but no joy